source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -706,6 +706,49 @@ namespace aspect
 
       return true;
     }
+
+    template <int dim>
+    std_cxx11::array<double,dim>
+    EllipsoidalChunk<dim>::cartesian_to_natural_coordinates(const Point<dim> &position_point) const
+    {
+      Assert(dim == 3,ExcMessage("This geometry model doesn't support 2d."));
+
+      // the chunk manifold works interally with a vector with longitude, latitude, depth.
+      // We need to output radius, longitude, latitude to be consistent.
+      const Point<dim> transformed_point = manifold.pull_back(position_point);
+
+      const double radius = get_radius(position_point);
+      std::array<double,dim> position_array;
+      position_array[0] = radius - transformed_point(2);
+      position_array[1] = transformed_point(1);
+      position_array[2] = transformed_point(0);
+
+      return position_array;
+    }
+
+
+
+    template <int dim>
+    Point<dim>
+    EllipsoidalChunk<dim>::natural_to_cartesian_coordinates(const std_cxx11::array<double,dim> &position_tensor) const
+    {
+      Assert(dim == 3,ExcMessage("This geometry model doesn't support 2d."));
+
+      Point<3> position_point;
+      position_point(0) = position_tensor[2];
+      position_point(1) = position_tensor[1];
+
+      const double radius = get_radius(reinterpret_cast<Point<dim>&>(position_point));
+
+      position_point(2) = radius - position_tensor[2];
```

Comment:
should this be `position_tensor[0]`? Above you have:
```
position_array[2] = transformed_point(0); 
```

Reply:
you are right, that does make more sense :)